cpp/ql/src/semmle/code/cpp/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -248,4 +249,18 @@ private predicate exprToPartialDefinitionStep(Expr exprIn, Expr exprOut) {
       exprIn = call.getArgument(argInIndex)
     )
   )
+  or
+  exists(Assignment a |
+    iteratorDereference(exprOut) and
+    a.getLValue() = exprOut and
+    a.getRValue() = exprIn
+  )
+}
+
+private predicate iteratorDereference(Call c) {
+    c.getTarget() instanceof IteratorArrayMemberOperator
+    or
+    c.getTarget() instanceof IteratorPointerDereferenceMemberOperator
+    or
+    c.getTarget() instanceof IteratorPointerDereferenceOperator
```

Comment:
We don't usually include an `implementations` file directly because it's very hard to keep code like this in sync with `Iterator.qll`. Can the property of being an iterator dereference become a model interface instead?